packages/flutter/lib/src/widgets/resizing_header_sliver.dart

```diff
@@ -0,0 +1,213 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:math' as math;
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/rendering.dart';
+
+import 'framework.dart';
+import 'slotted_render_object_widget.dart';
+
+/// A sliver that is pinned to the start of its [CustomScrollView] and
+/// reacts to scrolling by resizing between the intrinsic sizes of its
+/// min and max extent prototypes.
+///
+/// The minimum and maximum sizes of this sliver are defined by [minExtentPrototype]
+/// and [maxExtentPrototype], a pair of widgets that are laid out once. You can
+/// use [SizedBox] widgets to define the size limits.
+///
+/// This sliver is preferable to the general purpose [SliverPersistentHeader]
+/// for its relatively narrow use case because there's no need to create a
+/// [SliverPersistentHeaderDelegate] or to predict the header's minimum or
+/// maximum size.
+///
+/// {@tool dartpad}
+/// This sample shows how this sliver's two extent prototype properties can be used to
+/// create a resizing header whose minimum and maximum sizes match small and large
+/// configurations of the same header widget.
+///
+/// ** See code in examples/api/lib/widgets/sliver/resizing_header_sliver.0.dart **
+/// {@end-tool}
+class ResizingHeaderSliver extends StatelessWidget {
```

Comment:
If the minimum size defaulted to the child's size (instead of 0) then having this sliver effectively be a PinnedHeaderSliver would work. I'd still prefer to have a PinnedHeaderSliver because it's nice if the class says what you mean, although it could be implemented as a trivial wrapper for this one, or this one's implementation.

Reply:
I've made the minimum default 0 and the maximum default the intrinsic size of the child. So if you just specify a child you get a header that begins with the child's intrinsic size and shrinks to zero. If the same widget is provided for the minimum and maximum prototypes you get a pinned header.